### PR TITLE
Cortx 33182 multidata server pod

### PR DESCRIPTION
**Pull request by pragamj**
# Problem Statement
- [CORTX-33182] TA for Test test_44470 and RBF for test test_41909,test_41907

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
[TEST-N-NODE_TestContDeployment_20220720221511.log](https://github.com/Seagate/cortx-test/files/9156151/TEST-N-NODE_TestContDeployment_20220720221511.log)

-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)
[collections_log.txt](https://github.com/Seagate/cortx-test/files/9109587/collections_log.txt)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: config/prov/prov_test.yaml [hunk 1]
```diff
@@ -130,6 +130,8 @@ k8s_cortx_deploy:
   log_path: "/etc/cortx/log"
   custom_log_path: "/etc/cortx/custom/log"
   s3_max_start_timeout: 240
+  s3_instances_per_node: 3
```
Comment (priyankaborawake) [inline config/prov/prov_test.yaml:133]: Should it be 1 by default ?

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 2]
```diff
@@ -511,11 +513,11 @@ def update_sol_yaml(self, worker_obj: list, filepath: str, cortx_image: str,
             # The condition to validate the config.
             if not skip_disk_count_check and valid_disk_count > \
                     (cvg_count * node_list):
-                return False, "The sum of data disks per cvg " \
+                assert False, "The sum of data disks per cvg " \
```
Comment (priyankaborawake) [inline libs/prov/prov_k8s_cortx_deploy.py:516]: Do we want to assert in lib ?

**Comment by priyankaborawake** [comment]
Please attach execution logs for deployment with multipods. .

File: config/prov/prov_test.yaml (hunk 1 above)
Comment (pragamj) [reply config/prov/prov_test.yaml:133]: in solution.example.yaml file its mentioned as 3, so kept in the same way.

File: libs/prov/prov_k8s_cortx_deploy.py (hunk 2 above)
Comment (pragamj) [reply libs/prov/prov_k8s_cortx_deploy.py:516]: as the error message was very generic and we are not explicitly using this method in our test file so using assert to capture correct error message.

**Comment by pragamj** [comment]
@priyankaborawake, the execution is blocked as we haven't received the service release v0.9.0

File: config/prov/prov_test.yaml [hunk 3]
```diff
@@ -217,15 +219,15 @@ k8s_cortx_deploy:
         mem: '1Gi'
         cpu: '250m'
       limits:
-        mem: '3Gi'
+        mem: '2Gi'
```
Comment (priyankaborawake) [inline config/prov/prov_test.yaml:222]: Please add reference for these values.

File: tests/prov/test_prov_k8s_cortx_upgrade.py [hunk 4]
```diff
@@ -129,52 +127,24 @@ def get_status(self):
...
-        LOGGER.info("Installing CORTX image version: %s", upgrade_version)
-        if int(upgrade_version) <= int(installed_version):
-            assert False, "Installed version is same or higher than installing version."
-        else:
-            LOGGER.info("Installed version is lower than installing version.")
-        LOGGER.info("Step 4: Check cluster health.")
-        resp = self.upgrade_obj.prov_obj.check_s3_status(self.master_node_list[0], pod_prefix=
-        cons.POD_NAME_PREFIX)
-        assert_utils.assert_true(resp[0], resp[1])
+        # checkout the k8s repo tag
```
Comment (priyankaborawake) [inline tests/prov/test_prov_k8s_cortx_upgrade.py:130]: Remove unused code.

File: tests/prov/test_prov_k8s_cortx_upgrade.py [hunk 5]
```diff
@@ -129,52 +127,24 @@ def get_status(self):
...
                                                          self.upgrade_image)
         assert_utils.assert_true(resp[0], resp[1])
         LOGGER.info("Step 6: Check if installed version is equals to installing version.")
-        resp = HAK8s.get_config_value(self.master_node_list[0])
-        assert_utils.assert_true(resp[0], resp[1])
-        new_installed_version = resp[1]['cortx']['common']['release']['version'].split("-")[1]
-        LOGGER.info("New CORTX image version: %s", new_installed_version)
-        assert_utils.assert_equals(new_installed_version, upgrade_version,
-                                   "new_installed version is equal to upgrade_version.")
+        installed_version = self.upgrade_obj.prov_obj.get_installed_version(
```
Comment (priyankaborawake) [inline tests/prov/test_prov_k8s_cortx_upgrade.py:137]: Code doesn't match the description.

File: libs/prov/prov_k8s_cortx_deploy.py [hunk 6]
```diff
@@ -601,22 +605,15 @@ def update_nodes_sol_file(filepath, worker_obj):
         Param: worker_obj: list of node object
         :returns the filepath and status True
         """
-        node_list = len(worker_obj)
+        # node_list = len(worker_obj)
```
Comment (priyankaborawake) [inline libs/prov/prov_k8s_cortx_deploy.py:608]: Remove if not used.

File: tests/prov/test_prov_pods_deployment.py [hunk 7]
```diff
@@ -349,11 +354,49 @@ def test_41907(self):
...
+        self.collect_sb = False
+        self.destroy_flag = True
+        self.log.info("===Test Completed===")
+
+    @pytest.mark.lc
+    @pytest.mark.cluster_deployment
+    @pytest.mark.tags("TEST-44470")
+    def test_44470(self):
+        """
+        Test to verify the multi data and server pod being deployed with 1 cvg
```
Comment (priyankaborawake) [inline tests/prov/test_prov_pods_deployment.py:395]: Is this multi data pod test ?

File: config/prov/prov_test.yaml (hunk 1 above)
Comment (priyankaborawake) [inline config/prov/prov_test.yaml:133]: Will this resource allocation work with multiple data pods and multiple rgw instance ?

File: tests/prov/test_prov_k8s_cortx_upgrade.py [hunk 8]
```diff
@@ -215,6 +185,9 @@ def test_33669(self):
     def test_41953(self):
         """Verify resume of in progress rolling upgrade."""
         process_list = []
+        self.upgrade_obj.retain_solution_file(
```
Comment (priyankaborawake) [inline tests/prov/test_prov_k8s_cortx_upgrade.py:188]: Please rename the function retain_solution_file 
As it modifies the file to newer values and not save the original for further use.

Comment (priyankaborawake) [reply tests/prov/test_prov_k8s_cortx_upgrade.py:188]: Also there are multiple instance using the hardcoded 'solution.yaml'
Please add it as constants and then use it in libs, tests. 
can be found in retain_solution_file

**Review by priyankaborawake — CHANGES_REQUESTED** [review]

File: config/prov/prov_test.yaml (hunk 1 above)
Comment (pragamj) [reply config/prov/prov_test.yaml:133]: this is the default value provided in solution.yaml, so have kept that , for custom values user can pass that in the test of in jenkins job.

Comment (pragamj) [reply config/prov/prov_test.yaml:133]: For 3 and 5 these are working

File: config/prov/prov_test.yaml (hunk 3 above)
Comment (pragamj) [reply config/prov/prov_test.yaml:222]: Please refer the solution.example.yaml file in K8s repo.https://github.com/Seagate/cortx-k8s/blob/v0.9.0/k8_cortx_cloud/solution.example.yaml

**Review by priyankaborawake — APPROVED** [review]

**Review by ketanarlulkar — APPROVED** [review]

**Review by pranav-sahasrabudhe — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**